File: Domain/TimelineEvent/Model/TimelineEvent.php
<?php

declare(strict_types=1);

namespace Domain\TimelineEvent\Model;

use DateTime;
use Domain\Incident\Model\Incident;
use Domain\User\Model\User;

class TimelineEvent
{
    private string $id;
    private TimelineEventType $type;
    private string $message;
    private DateTime $occurredAt;
    private bool $raisedBySystem;
    private Incident $incident;
    private ?User $author = null;

    public function __construct(
        string $id,
        TimelineEventType $type,
        string $message,
        DateTime $occurredAt,
        bool $raisedBySystem,
        Incident $incident,
        ?User $author = null
    ) {
        $this->id = $id;
        $this->type = $type;
        $this->message = $message;
        $this->occurredAt = $occurredAt;
        $this->raisedBySystem = $raisedBySystem;
        $this->incident = $incident;
        $this->author = $author;
    }

    public function getId(): string
    {
        return $this->id;
    }

    public function getType(): TimelineEventType
    {
        return $this->type;
    }

    public function getMessage(): string
    {
        return $this->message;
    }

    public function getOccurredAt(): DateTime
    {
        return $this->occurredAt;
    }

    public function isRaisedBySystem(): bool
    {
        return $this->raisedBySystem;
    }

    public function getIncident(): Incident
    {
        return $this->incident;
    }

    public function getAuthor(): ?User
    {
        return $this->author;
    }
}

File: src/main/java/com/uptime/incident/domain/model/TimelineEvent.java
package com.uptime.incident.domain.model;

import com.uptime.incident.domain.model.enums.TimelineEventType;

import java.time.Instant;

public class TimelineEvent {
    private TimelineEventType timelineEventType;
    private String message;
    private Instant occurredAt;
    private Boolean raisedBySystem;

    public TimelineEventType getTimelineEventType() {
        return timelineEventType;
    }

    public void setTimelineEventType(TimelineEventType timelineEventType) {
        this.timelineEventType = timelineEventType;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Instant getOccurredAt() {
        return occurredAt;
    }

    public void setOccurredAt(Instant occurredAt) {
        this.occurredAt = occurredAt;
    }

    public Boolean getRaisedBySystem() {
        return raisedBySystem;
    }

    public void setRaisedBySystem(Boolean raisedBySystem) {
        this.raisedBySystem = raisedBySystem;
    }
}
